provides.py

```diff
@@ -38,14 +38,23 @@ def joined_or_changed(self):
         if self._has_auth_request():
             conv.set_state('{relation_name}.auth.requested')
 
-    @hook('{provides:kube-control}-relation-{broken,departed}')
+    @hook('{provides:kube-control}-relation-departed')
     def departed(self):
         """Remove all states.
 
         """
         conv = self.conversation()
         conv.remove_state('{relation_name}.connected')
         conv.remove_state('{relation_name}.gpu.available')
+        conv.set_state('{relation_name}.departed')
+
+    def flush_departed(self):
+        """Remove the signal state that we have a unit departing the
+        relationship. Additionally return the unit departing so the host can
+        do any cleanup logic required. """
+        conv = self.conversation()
+        conv.remove_state('{relation_name}.departed')
+        return conv.scope
```

Comment:
That is possible because scope is set to UNIT?

I did not know this was even possible! Really nice!